### ISSUE TITLE
ENH: Implement PVSyst Cell Temperature Model

### ISSUE THREAD

**Issue by mikofski** [pvlib/pvlib-python#552]
**PVSyst Cell Temperature Model**
For users using PVSyst model chain, there is no equivalent cell temperature model. The only cell temperature model is the Sandia array performance model, [`pvlib.pvsystem.sapm_celltemp`](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.sapm_celltemp.html). The [PVSyst cell temperature model](https://pvpmc.sandia.gov/modeling-steps/2-dc-module-iv/cell-temperature/pvsyst-cell-temperature-model/) is different.

**Describe the solution you'd like**
Add a corresponding method like `pvlib.pvsystem.pvsyst_celltemp`. I'll bet it's already implemented in PVLIB_MATLAB, right?

**Describe alternatives you've considered**
leave it to the user to implement

```python
def pvsyst_celltemp(ambient_temp, gti, panel_efficiency, windspeed, alpha_absorption=0.9,
                    natural_convenction_coeff=29, forced_convection_coeff=0):
    """
    Calculate PVSyst cell temperature.

    Parameters
    ------------
    ambient_temp : numeric
        Ambient temperature in degrees Celsius
    gti : numeric
        Global tilted irradiance, AKA incident irradiance in the plane of array (POA) in W/m^2
    panel_efficiency : numeric
        PV module efficiency as a fraction
    windspeed : numeric
        Wind speed in m/s
    alpha_absorption : numeric
        Absorption coefficient, default is 0.9
    natural_convenction_coeff : numeric
        Natural convection coefficient, default is 29
    forced_convection_coeff : numeric
        Forced convection coefficient, default is zero

    Returns
    --------
    celltemp : numeric
        Cell temperature in degrees Celsius
    """
    combined_convection_coeff = (
        natural_convenction_coeff + forced_convection_coeff * windspeed
    )
    return (ambient_temp
        + alpha_absorption * gti * (1 - panel_efficiency) / combined_convection_coeff)
```

I'm not married to these names, they are only for demonstration

**Additional context**
* is this related to #487 at all? I tried to look for any corresponding or existing issues
* if this cell temperature model is implmented then should others be as well? what does SAM use?

**Comment by cwhanse** [comment]
Great proposal @mikofski.  The [PVSyst temperature model](http://files.pvsyst.com/help/index.html) is   an extension of the [Faiman ](https://pvpmc.sandia.gov/modeling-steps/2-dc-module-iv/module-temperature/faiman-module-temperature-model/) temperature model. We don't have either in PVLib MATLAB. They are easy enough to code but I don't know of a database of coefficients. Like the Sandia model, coefficient values depend both on the module construction and its mounting. PVSyst provides some defaults.

I read the recent bulletin board exchange between Mike Anderson and Andre Mermoud, it appears that there is still uncertainty regarding some of the PVsyst model terms (efficiency).

The Faiman model is prescribed by at least one IEC standard so it is of broad interest.  I would like to see this model added as well.  Different pull request, though.

How do people feel about changing the naming convention: `celltemp_sapm`, `celltemp_pvsyst` etc. rather than `sapm_celltemp` etc.?

**Comment by wholmgren** [comment]
At the risk of issue scope creep... what about a ``celltemp.py`` module with functions ``sapm``, ``pvsyst``, etc.?

**Comment by cwhanse** [comment]
Works for me

Sent from my iPhone

On Aug 31, 2018, at 4:27 PM, Will Holmgren <notifications@github.com<mailto:notifications@github.com>> wrote:


At the risk of issue scope creep... what about a celltemp.py module with functions sapm, pvsyst, etc.?

**Comment by wholmgren** [comment]
Closed by #628.

**Comment by adriesse** [comment]
I have a PR coming...
